### ISSUE TITLE
Sharing DB link for a .gif file returns static image

### ISSUE THREAD

**Issue by Benjamski** [fniephaus/alfred-dropbox#2]
I'm very likely not understanding exactly what this workflow is copying to the clipboard but I'm trying to build my own workflow that simply accesses one folder (within public directory) of my drop box, queries by filename, copies to clipboard the public dropbox link for the file on return and then pastes it into the current cursor location .

I've got everything but the last two working just fine. I always get a broken link on my workflow and when I used yours the shared link only returns a static image of the gif.

I know yours is directly using the DB API whereas I think what I'm trying to do is locally access the files and somehow copy the public link (as you would by right-clicking the file in finder) and then paste it, but one step at a time of course.

Feel free to just close this, but I've seen and used a lot of your workflows and I'd appreciate any tips you could send my way!

**Comment by fniephaus** [comment]
Hi @Benjamski,
I'm not exactly sure I understand what your are trying to do and what problem you are running into.
Could you please elaborate a bit more? What are you trying to achieve with your workflow?

**Comment by Benjamski** [comment]
Sure! @fniephaus It's really quite simple, but I've had much more experience with using alfred workflows than making my own. I haven't got far on my own but found [this](http://destroytoday.com/writings/gif-workflow/) which is doing almost the very same thing. I've tried following this along and replacing my unique ID but I always get a broken link.

```
dropbox_url="http://dl.dropboxusercontent.com/u/5056567/gifs/"
echo -ne ${dropbox_url}$(basename {query}) | pbcopy
```

I have this triggering on a keyword, with the search scope of the correct folder but it must not be copying the link correctly.

**Comment by fniephaus** [comment]
Ah, I think I understand. This is looks in a folder and adds the filename to your public folder's base address. So that's why you don't need authentication nor talking to Dropbox's API.

Let's say there's a file in your gifs folder called `my.gif`.
Do you get an error when visiting http://dl.dropboxusercontent.com/u/5056567/gifs/my.gif ?

If you do, there's something wrong with either the dropbox_url or the way you have set up your folder locally. It has to be in the Public folder. Thing is, Dropbox has deprecated the Public folder (see [here](https://www.dropbox.com/en/help/16)), so I'm afraid all of this doesn't work if you opened your DB account after October 4, 2012.

**Comment by Benjamski** [comment]
Ahh. I was unsure when my account was created, but in my security settings I can see the oldest device used at 5 years ago, so I think I beat the cutoff. 

Using that link for example, with an actual file in my gifs folder I do get the link  working properly  https://dl.dropboxusercontent.com/u/5056567/gifs/dontTouchHorse.gif

Am I just not copying it correctly?

**Comment by fniephaus** [comment]
What link do you get when selecting this gif?

**Comment by Benjamski** [comment]
This is the link i get when successfully querying it from the Alfred workflow:

http://dl.dropboxusercontent.com/u/5056567/gifs/gifs/dontTouchHorse.gif

I also saw a similar workflow with the following script:
```
dropbox_url="https://www.dropbox.com/sh/234la0tzlzg8bls/AACDFIxvzolOPLCzZv8BDF8za?dl=0" 
gif_path=`dirname {query}`
gif_folder=$( printf "%s\n" "$(basename "${gif_path}")" | sed 's/ /%20/g' )
echo -ne ${dropbox_url}${gif_folder}/$(basename {query}) | pbcopy
```


http://dl.dropboxusercontent.com/u/5056567/gifs/gifs/dontTouchHorse.gif

**Comment by fniephaus** [comment]
What happens if you set your `dropbox_url` to just `dropbox_url="http://dl.dropboxusercontent.com/u/5056567/"`?

**Comment by Benjamski** [comment]
hm I get a 404 with that as the dropbox_url. When I paste the copied code from that it comes out as

```
http://dl.dropboxusercontent.com/u/5056567/my.gif
```

**Comment by Benjamski** [comment]
Aha I figured it out! So I had two issues:
-There were some gifs in child directories, so the path was incomplete whenever I tested one of those.
-The id number is the unique id to my "Public" folder, but the gifs are one node deeper in /gifs/ so I did need that.

Here's the working script:
```
dropbox_url="https://dl.dropboxusercontent.com/u/5056567/gifs/"
echo -ne ${dropbox_url}$(basename {query}) | pbcopy
```

I apologize for opening such a simple issue for this, Im very new to programming and looking at this thread I'm thinking that I just didn't test it on my own enough to find such a simple fix.

Thank you so much for helping me out anyway, I really appreciate it!
